### PR TITLE
Short Hash

### PR DESCRIPTION
**Pull request by JCOR11599**
hashes can't contain any letters past 'v', as they don't encode/decode correctly.

### CODE REVIEW

File: go/store/nbs/table_index.go [hunk 1]
```diff
@@ -311,6 +312,119 @@ func (ti onHeapTableIndex) Prefixes() ([]uint64, error) {
...
+	// Pad string
+	for i := len(s); i < 16; i++ {
+		s += p
+	}
+	// Decode
+	h, _ := encoding.DecodeString(s)
+	return binary.BigEndian.Uint64(h)
+}
+
+func (ti onHeapTableIndex) ResolveShortHash(short string) ([]string, error) {
```
Comment (andrew-wm-arthur) [inline go/store/nbs/table_index.go:377]: this method should take a `[]byte` not a string

File: go/store/nbs/table_index_test.go [hunk 2]
```diff
@@ -102,3 +102,23 @@ func TestMMapIndex(t *testing.T) {
 	assert.Equal(t, idx.TableFileSize(), mmidx.TableFileSize())
 	assert.Equal(t, idx.TotalUncompressedData(), mmidx.TotalUncompressedData())
 }
+
+func TestOnHeapTableIndex_ResolveShortHash(t *testing.T) {
```
Comment (andrew-wm-arthur) [inline go/store/nbs/table_index_test.go:106]: I'd like to see some more comprehensive testing of the resolve method. Specifically, you should construct your own index with a known set of hashes, and then test resolving short hashes of different lengths.
Check out `nbs/table_writer.go` to construct an index from scratch

**Review by andrew-wm-arthur — COMMENTED** [review]
Looks like you're on the right track, but it needs thorough test coverage

**Comment by VinaiRachakonda** [comment]
#benchmark

File: go/store/nbs/table_index.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply go/store/nbs/table_index.go:393]: you should refactor this to take a `[]byte`

**Review by andrew-wm-arthur — APPROVED** [review]

**Comment by andrew-wm-arthur** [comment]
LGTM

**Pull request merged**